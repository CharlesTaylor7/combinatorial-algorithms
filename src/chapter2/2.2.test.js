import { binaryReflect, nthGrayCode, allGrayCodes } from './2.2'

test('binaryReflect', () => {
  expect(binaryReflect(['0', '1']))
    .toEqual(['00','01', '11', '10'])
})

test('G(0)', () => {
  expect(nthGrayCode(0)).toEqual([''])
})

test('G(1)', () => {
  expect(nthGrayCode(1)).toEqual(['00', '01'])
})

test('G(2)', () => {
  expect(nthGrayCode(2)).toEqual(['00', '01', '11', '10'])
})

test('G(3)', () => {
  expect(nthGrayCode(3)).toEqual([
    '000', '001', '011', '010',
    '110', '111', '101', '100',
  ])
})

test('all gray codes of order 4', () => {
  expect(allGrayCodes(4)).toMatchSnapshot();
})
