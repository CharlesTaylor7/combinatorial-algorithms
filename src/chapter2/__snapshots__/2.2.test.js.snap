// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all gray codes of order 4 1`] = `
Wu {
  "next": [Function],
}
`;
